exec/plugin-commands-script-runners/src/ciLogs.ts

```diff
@@ -0,0 +1,61 @@
+import CI, { isCI } from 'ci-info'
+
+export class CiLogs {
+  constructor (private readonly opts: {
+    concurrency?: number
+    script?: string
+    name?: string
+    version?: string
+    prefix: string
+  }) {
+    if (!this.opts.concurrency) {
+      this.opts.concurrency = 4
+    }
+  }
+
+  private labels () {
+    const { name, version, prefix, script } = this.opts
+    const ScriptName = `${name ?? 'unknown'}${version ? `@${version}` : ''} ${script ? `: ${script}` : ''} ${prefix}`
+
+    if (CI.GITHUB_ACTIONS) {
+      return {
+        start: `::group::${ScriptName}\r\n`,
+        end: '::endgroup::\r\n',
+      }
+    } else if (CI.GITLAB) {
+      return {
+        start: `section_start:${Math.floor(Date.now() / 1000)}:${prefix}\\r\\e[0K${ScriptName}`,
+        end: `section_end:${Math.floor(Date.now() / 1000)}:${prefix}\\r\\e[0K`,
+      }
+    } else if (CI.TRAVIS) {
+      return {
+        start: `travis_fold:start:${prefix}${ScriptName}\r\n`,
+        end: `travis_fold:end:${prefix}${ScriptName}\r\n`,
+      }
```

Comment:
why is prefix printed twice for gitlab and travis?

Reply:
True, on it

Reply:
Same for Gitlab I think, even though the prefix itself is not printed as it's an ID, I'm going to add a property for ID stuff

Reply:
can't we just duplicate "script name" as id then?